src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -97,6 +105,27 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
                     contentType = "application/json";
                 }
             }
+            else if (isBinaryDataSet)
+            {
+                // As per the spec, if the implementation determines that the type of data is Binary,
+                // the value MUST be represented as a JSON string expression containing the Base64 encoded
+                // binary value, and use the member name data_base64 to store it inside the JSON object.
+                var decodedBody = Encoding.UTF8.GetString(binaryData.GetBytesFromBase64());
```

Comment:
It's not safe to do `Encoding.UTF8.GetString(...)` on the return value here, the point is that it's *just bytes* and you can't make any assumptions about the format.